sklearn/utils/validation.py

```diff
@@ -444,7 +492,7 @@ def check_X_y(X, y, accept_sparse=None, dtype="numeric", order=None, copy=False,
         y = column_or_1d(y, warn=True)
         _assert_all_finite(y)
     if y_numeric and y.dtype.kind == 'O':
-        y = y.astype(np.float64)
+        y = astype(y, np.float64, copy=False)
```

Comment:
This specific change can also be reverted as here `y` is guaranteed to be an array of objects so `copy=False` will never bring anything.